pkg/cluster/cluster.go

```diff
@@ -403,6 +377,66 @@ func (c *Cluster) compareStatefulSetWith(statefulSet *v1beta1.StatefulSet) *comp
 	return &compareStatefulsetResult{match: match, reasons: reasons, rollingUpdate: needsRollUpdate, replace: needsReplace}
 }
 
+type ContainerCheck struct {
+	condition func(a, b v1.Container) bool
+	reason    string
+}
+
+// compareContainers: compare containers from two stateful sets
+// and return:
+// * whether or not roll update is needed
+// * a list of reasons in a human readable format
+func (c *Cluster) compareContainers(setA, setB *v1beta1.StatefulSet) (bool, []string) {
```

Comment:
This function could just have been copy & paste from the function above and it would have helped or?

Transforming 20 lines into 50 lines does not immediately convince me that the simple "if" is not good enough.

Reply:
Could have maybe looked shorter if it would have been just:

checks.append("container name does not match"lambda x,y: x.name != y.name)
checks.append(...)

Reply:
Number of lines grew mostly because of formatting stuff, I don't think it's a valid metric here.
I also don't think that it's possible to make it shorter, but I can easily return the implementation
with if conditions. Even this would be better than it's now.